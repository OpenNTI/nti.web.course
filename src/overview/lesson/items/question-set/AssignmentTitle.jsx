import './AssignmentTitle.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';

const DEFAULT_TEXT = {
	assignmentTitle: {
		title: '%(title)s',
		points: '%(points)spts.'
	}
};
const t = scoped('course.overview.lesson.overview.question-set.AssignmentTitle', DEFAULT_TEXT);


LessonOverviewAssignmentTitle.propTypes = {
	assignment: PropTypes.object
};
export default function LessonOverviewAssignmentTitle ({assignment}) {
	const {title} = assignment;

	return (
		<span className="lesson-overview-assignment-title">
			<span className="title">{t('assignmentTitle.title', {title})}</span>
		</span>
	);
}
