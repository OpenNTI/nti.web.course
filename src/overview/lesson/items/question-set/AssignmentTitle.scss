.lesson-overview-assignment-title {
	display: flex;
	align-items: center;
	max-width: 100%;

	.title {
		flex: 0 1 auto;
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
		padding-right: 0.3rem;
	}

	.points {
		flex: 0 0 auto;
		font-weight: 600;
	}
}
