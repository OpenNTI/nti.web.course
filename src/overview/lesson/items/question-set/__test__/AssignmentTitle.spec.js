import React from 'react';
import renderer from 'react-test-renderer';

import AssignmentTitle from '../AssignmentTitle';

/* eslint-env jest */
describe('Course overview assignment title test', () => {
	test('Title only', async () => {
		const assignment = {
			title: 'My Assignment'
		};

		const cmp = renderer.create(
			<AssignmentTitle assignment={assignment}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Title with totalPoints', async () => {
		const assignment = {
			title: 'My Assignment',
			totalPoints: 5
		};

		const cmp = renderer.create(
			<AssignmentTitle assignment={assignment}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
