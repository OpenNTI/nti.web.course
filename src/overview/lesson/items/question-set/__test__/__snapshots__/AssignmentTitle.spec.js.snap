// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course overview assignment title test Title only 1`] = `
<span
  className="lesson-overview-assignment-title"
>
  <span
    className="title"
  >
    My Assignment
  </span>
</span>
`;

exports[`Course overview assignment title test Title with totalPoints 1`] = `
<span
  className="lesson-overview-assignment-title"
>
  <span
    className="title"
  >
    My Assignment
  </span>
</span>
`;
